### PR TITLE
build: release 0.1.0-alpha.0

### PR DESCRIPTION
**Pull request by johnyob**
# Context

# Description

This PR:
- Updates all the `Cargo.toml`'s accordingly with the desired version + added more workspace globals (e.g. homepage, description, etc)
- Changes to jstz-cli:
   - Added detection to whether jstz is running in a container (if so disables: `jstz sandbox stop`, `jstz sandbox restart` and `jstz docs`). 
   - Fixed rollup node bug with `--history-mode archive`
   - Changed paths for jstz sandbox files to include `/usr/share/jstz`
- Added CI/CD pipeline for deploying to weeklynet and building docker containers for Linux (amd,arm) + MacOS M1
- Updated jstz bridge to let users mint ctez (some changes due to updated version of ligo and changes in compiled types)

# Manually testing the PR

### CODE REVIEW

**Comment by johnyob** [comment]
> Can you please add a list of changes + manual testing guide to this PR as it seems to make many different changes?

Added list of changes. Manual testing is hard to do since this isn't really a feature change. But I'd check the sandbox + the bridge for testing.

**Comment by johnyob** [comment]
> (also it is obviously your choice, but I would generally prefer if we avoided making changes on sunday)

Agreed. I only did this so avoid waiting a bunch on our docker CI on Monday (since it is terribly slow)

**Review by ryutamago — COMMENTED** [review]
I still get the error `No such file or directory (os error 2)` when running the sandbox

**Review by ryutamago — APPROVED** [review]
> I still get the error `No such file or directory (os error 2)` when running the sandbox

this was because my `config.json` file was incorrect, the sandbox works fine. We should probably improve the error message for the config file.

Great job! 🛳️

**Review by alanmarkoTrilitech — CHANGES_REQUESTED** [review]
Overall lgtm, just some minor issues

File: contracts/jstz_ctez.mligo [hunk 1]
```diff
@@ -60,14 +60,11 @@ module Jstz_ctez = struct
     set_balance s address (f (get_balance s address))
 
   [@entry] let mint (mint : mint) (s : storage) : return = 
-    if Tezos.get_sender () <> s.admin
-    then failwith "Only `admin` can mint tokens"
```
[the remark this answers is not in the document]
Comment (johnyob) [reply contracts/jstz_ctez.mligo:64]: It permits people to manually mint ctez on the weeklynet deployment

File: crates/jstz_cli/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+COPY --from=octez /usr/local/bin/octez-smart-rollup-node /usr/bin/octez-smart-rollup-node
+COPY --from=octez /usr/local/bin/octez-node /usr/bin/octez-node
+COPY --from=octez /usr/local/bin/octez-client /usr/bin/octez-client
+COPY --from=octez /usr/share/zcash-params /root/.zcash-params
+# Copy the jstz binary & dependencies
+COPY --from=builder /target/debug/jstz /usr/bin/jstz
+COPY --from=builder /crates/jstz_cli/jstz_kernel.wasm /usr/share/jstz/jstz_kernel.wasm
+COPY --from=builder /crates/jstz_cli/sandbox-params.json /usr/share/jstz/sandbox-params.json
+COPY --from=builder /crates/jstz_cli/sandbox.json /usr/share/jstz/sandbox.json
+ENTRYPOINT [ "/usr/bin/jstz" ]
```
[the remark this answers is not in the document]
Comment (johnyob) [reply crates/jstz_cli/Dockerfile:27]: I'd disagree, the container shouldn't ship with examples

File: crates/jstz_cli/src/bridge/deposit.rs [hunk 3]
```diff
@@ -46,9 +46,9 @@ pub fn exec(
             "jstz_bridge",
             "deposit",
             &format!(
-                "(Pair {} 0x{})",
+                "(Pair 0x{} {})",
+                hex::encode_upper(to_pkh.as_bytes()),
                 amount,
-                hex::encode_upper(to_pkh.as_bytes())
```
[the remark this answers is not in the document]
Comment (johnyob) [reply crates/jstz_cli/src/bridge/deposit.rs:51]: LIGO changes the order that their tuple types are compiled down to Michelson pairs (as part of the LIGO v1 release) -- which means any Michelson pair we manually construct needs updating

File: crates/jstz_cli/src/sandbox/daemon.rs [hunk 4]
```diff
@@ -627,6 +634,10 @@ pub async fn main(detach: bool, background: bool, cfg: &mut Config) -> Result<()
     }
 
     if detach {
+        if in_container() {
+            bail_user_error!("Detaching from the terminal is not supported in this environment. Please run `jstz sandbox start` without the `--detach` flag.");
```
[the remark this answers is not in the document]
Comment (johnyob) [reply crates/jstz_cli/src/sandbox/daemon.rs:638]: Yes, but not worth it for this PR (would require waiting for CI to rebuild the containers)

File: crates/jstz_cli/build.rs [hunk 5]
```diff
@@ -16,30 +16,48 @@ const SANDBOX_PARAMS_PATH: &str = "./sandbox-params.json";
 
 const SANDBOX_PATH: &str = "./sandbox.json";
 
-fn generate_path_getter_code(name: &str, path: &Path) -> String {
+fn generate_path_getter_code(out_dir: &Path, name: &str, file: &str) -> String {
```
[the remark this answers is not in the document]
Comment (johnyob) [reply crates/jstz_cli/build.rs:19]: It generates the Rust code that gets the path for various static files

**Review by alanmarkoTrilitech — APPROVED** [review]
LGTM :ship:

**Pull request merged**